Combo box v1 (#1963)
jpmorganchase/salt-ds: packages/lab/stories/combo-box-next/combo-box-next.doc.stories.mdx

import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
import { ComboBoxNext } from "@salt-ds/lab";
import {
  CharacteristicUsage,
  CSSClassTable,
} from "css-variable-docgen-components";

<Meta title="Documentation/Lab/Combo Box Next" component={ComboBoxNext} />

# Combo Box

Combo Box helps users select an item from a large list of options without scrolling. The typeahead functionality makes this selection quicker, easier, and reduces errors.

Users can see a list of available options when they click on the component and filter the list as they type. Once they’ve made their selection, it populates the field and the overlay list closes.

<Canvas>
  <Story id="lab-combo-box-next--default" />
</Canvas>

## How to use

### When to use Combo Box

> You should use Combo Box when you have a list of 10 or more items, that don’t need to be permanently visible to the user like they would be in a filter panel or form.

Reviewer: Phrasing Suggestion: 
Combo Box should be used when you have a list of 10 or more items, that don’t need to be permanently visible like they would be in a filter panel or form.

### When not to use Combo Box

> There may be occasions where a different component would suit your needs better, for Combo Box these could include:

Reviewer: Can probs remove this line

- If you have a short list of five to 10 options that don’t require filtering, use a `Dropdown`.
- If you have fewer than five options, use `Checkbox` or `Radio Button` groups.

## Features

### Single select

A single item can be selected from the Combo Box options.

<Canvas>
  <Story id="lab-combo-box-next--default" />
</Canvas>

### Customized renderer and filter

You can give users a quicker and easier experience when they’re choosing their options by changing the logic of the component’s typeahead filter. In this example, you can define a filter that matches country and code.

> Occasionally, your users will benefit from seeing a particular visual with your overlay list items, so you can customize the display as you see fit. For example, if displaying a list of countries it may help to provide flags.

Reviewer: Phrasing Suggestion:
The typeahead filter logic can be modified to assist the user selection. This example filters on both the country and code.
Additionally the display of list items can be customised to visually assist users, like the rendering flag icons next to the name of countries seen below.

import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
import { ComboBoxNext } from "@salt-ds/lab";
import {
  CharacteristicUsage,
  CSSClassTable,
} from "css-variable-docgen-components";

> <Meta title="Documentation/Lab/ComboBox Next" component={ComboBoxNext} />

Reviewer 2: ```suggestion
<Meta title="Documentation/Lab/Combo Box Next" component={ComboBoxNext} />
```

> # Combobox

Reviewer 2: ```suggestion
# Combo Box
```

Combo Box helps users select an item from a large list of options without scrolling. The typeahead functionality makes this selection quicker, easier, and reduces errors.

Users can see a list of available options when they click on the component and filter the list as they type. Once they’ve made their selection, it populates the field and the overlay list closes.

<Canvas>
  <Story id="lab-combo-box-next--default" />
</Canvas>

## How to use

### When to use Combo Box

You should use Combo Box when you have a list of 10 or more items, that don’t need to be permanently visible to the user like they would be in a filter panel or form.

> ### When not to use List

Reviewer 3: ```suggestion
### When not to use Combo Box
```

There may be occasions where a different component would suit your needs better, for Combo Box these could include:

- If you have a short list of five to 10 options that don’t require filtering, use a `Dropdown`.
- If you have fewer than five options, use `Checkbox` or `Radio Button` groups.

## Features

### Single select

<Canvas>
  <Story id="lab-combo-box-next--default" />
</Canvas>

### Customized renderer and filter

<Canvas>
  <Story id="lab-combo-box-next--custom-renderer" />
</Canvas>

### Variants

<Canvas>
  <Story id="lab-combo-box-next--secondary" />
</Canvas>

# API

```
import { ComboBoxNext } from "@salt-ds/lab";
```

## Props

<ArgsTable of={ComboBoxNext} />

## Classes

<CSSClassTable of={ComboBoxNext} />

## Characteristics used

<CharacteristicUsage of={ComboBoxNext} />

## Accessibility

> List Next is accessible by assistive technologies like screen readers, and

Reviewer 3: ```suggestion
Combo Box Next is accessible by assistive technologies like screen readers, and
```
